src/geom/density.jl

```diff
@@ -0,0 +1,191 @@
+struct DensityGeometry <: Gadfly.GeometryElement
+    stat::Gadfly.StatisticElement
+    order::Int
+    tag::Symbol
+end
+
+function DensityGeometry(; n=256,
+                           bandwidth=-Inf,
+                           adjust=1.0,
+                           kernel=Normal,
+                           trim=false,
+                           scale=:area,
+                           position=:dodge,
+                           orientation=:horizontal,
+                           order=1,
+                           tag=empty_tag)
+    stat = Gadfly.Stat.DensityStatistic(n, bandwidth, adjust, kernel, trim,
+                                        scale, position, orientation, false)
+    DensityGeometry(stat, order, tag)
+end
+
+DensityGeometry(stat; order=1, tag=empty_tag) = DensityGeometry(stat, order, tag)
+
+"""
+   Geom.density(; bandwidth, adjust, kernel, trim, scale, position, orientation, order)
+
+Draws a kernel density estimate. This is a cousin of [`Geom.histogram`](@ref)
```

Comment:
it would be nice to document how Geom.density(Stat.identity) behaved.  in other words, what aesthetics does Geom.density directly use.

Reply:
Good point, I'll add this.